mvt/layer.go

```diff
@@ -139,38 +139,20 @@ func (l *Layer) Features() (f []Feature) {
 	return f
 }
 
-//AddFeatures will add one or more Features to the Layer, if a features ID is a the same as
-//Any already in the Layer, it will ignore those features.
-//If the id fields is nil, the feature will always be added.
-func (l *Layer) AddFeatures(features ...Feature) (skipped bool) {
-
+// AddFeatures will add one or more Features to the Layer
+// per the spec features SHOULD have unique ids but it's not required
+func (l *Layer) AddFeatures(features ...Feature) {
+	// pre allocate memory
 	b := make([]Feature, len(l.features), len(l.features)+len(features))
+
 	copy(b, l.features)
+
 	l.features = b
```

Comment:
I would redo this to the following:

```go
// per-allocate memory
b := make([]Feature, len(l.features)+len(features))
copy(b, l.features)
copy(b[len(l.features):],features)
l.features = b
```

copy is faster then append if you just need to copy over memory.